src/org/intellij/erlang/psi/impl/ErlangPsiImplUtil.java

```diff
@@ -1839,38 +1875,110 @@ public ErlangFunctionCallParameter(@NotNull String funName, @NotNull String modu
 
     @Override
     public boolean accepts(@NotNull T element, ProcessingContext context) {
-      ErlangExpression expr = PsiTreeUtil.getParentOfType(element, ErlangExpression.class, false);
-      if (expr == null) return false;
-      PsiElement list = expr.getParent();
-      if (list instanceof ErlangArgumentList) {
-        PsiElement funCall = list.getParent();
-        List<ErlangExpression> expressions = ((ErlangArgumentList) list).getExpressionList();
-        if (!(expressions.size() > myPosition && expressions.get(myPosition) == expr)) return false;
-        if (funCall instanceof ErlangFunctionCallExpression) {
-          PsiReference reference = funCall.getReference();
-          // option for resolve/multi resolve
-          ResolveResult[] results = reference instanceof PsiPolyVariantReference ?
-            ((PsiPolyVariantReference) reference).multiResolve(true) : ResolveResult.EMPTY_ARRAY;
-          for (ResolveResult r : results) {
-            if (expectedFunction(r.getElement())) return true;
+      ErlangExpression argument = getFunctionCallArgument(element);
+      if (argument == null || getPositionInFunctionCall(argument) != myPosition) return false;
```

Comment:
Can't this logic be made more explicit? It says "functionCallArgument", "positionInFunctionCall" - yet we don't see actual function call anywhere around. Will it be easy to debug this code? I doubt that.